python/lsst/meas/extensions/multiprofit/utils.py

```diff
@@ -35,20 +37,32 @@ def defaultdictNested():
 
 
 # TODO: Allow addition to existing image
-def get_spanned_image(footprint, bbox=None):
-    spans = footprint.getSpans()
+def get_spanned_image(footprint, bbox=None, spans=None, exposure=None, get_sig_inv: bool = False):
```

Comment:
I highly recommend making use of the `SpanSet.asArray` method and doing most of this using numpy arrays as opposed to using footprints, spansets, etc. It will be easier to read and probably run faster.